src/Services/Qrda/Helpers/Cat1View.php

```diff
@@ -152,15 +152,19 @@ public function result_value_as_string($result)
         }
 
         $oid = $result['system'] ?? $result['codeSystem'];
-        $system = $this->get_code_system_for_oid($oid);
-        if (!empty($result['code'])) {
-            return "<value xsi:type=\"CD\" code=\"" . $result['code'] . "\" codeSystem=\"" . $oid
-                . "\" codeSystemName=\"" . $system . "\"/>";
-        } elseif (!empty($result['value'])) {
-            return "<value xsi:type=\"PQ\" value=\"" . $result['value'] . "\" unit=\"" . ($result['unit'] ?: "UNK") . "\"/>";
+        if (!empty($oid)) {
+            $system = $this->get_code_system_for_oid($oid);
+            if (!empty($result['code'])) {
```

Comment:
This won't work in many resulting cases(labs).  There are results like urine color 'YELLOW' e.t.c. that will not have an oid.

Reply:
If it's allowed to be empty, then we can't pass an $oid to get_code_for system, and the rest of this code needs to be checked. It was crashing my system.

Reply:
Then I guess filter the oid with null coalesce or condition as long as a valid result is not filtered out. This came up several times in import.